### DISCUSSION TITLE
`npm-shrinkwrap.json` is too large, could be zipped (compressed)

### DISCUSSION THREAD

**Discussion by RupertBarrow** [npm/feedback#1111] [CLI]
### Statement
When an npm-shrinkwrap.json file is included in a module, this makes it quite large.
Here is a real world example : when creating small `oclif` plugins, each plugin (`npm` module) has its own `npm-shrinkwrap.json` file which, when added together, count for up to 60% of the total size of the resulting app.

### Solution
It would be nice to have an option to generate a compressed version of `npm-shrinkwrap.json` instead of the expanded version; this could be called `npm-shrinkwrap.gz`, for example.
The `npm` tooling would also be able to read these compressed files without any external preliminary decompression.

**Comment by ljharb** [comment]
It’s already compressed in the tarball. I think npm doesn’t need it after installing, so you could probably remove all shrinkwrap files inside node_modules postinstall, if the disk space is an issue.

(also, the app at dev time is always larger than at runtime; in production, there shouldn’t be anything in node_modules that isn’t used)

**Reply by RupertBarrow** [reply]
Thanks @ljharb. We're talking about developing plugins here. I'm not authoring them, I'm using them (cf https://github.com/forcedotcom/cli/discussions/2835)
